packages/core/src/subscriptions/index.ts

```diff
@@ -0,0 +1,342 @@
+import { Bundle, Parameters, Subscription, SubscriptionStatus } from '@medplum/fhirtypes';
+import { MedplumClient } from '../client';
+import { TypedEventTarget } from '../eventtarget';
+import { OperationOutcomeError, serverError, validationError } from '../outcomes';
+import { ProfileResource, getReferenceString, resolveId } from '../utils';
+
+export type SubscriptionEventMap = {
+  connect: { type: 'connect'; payload: { subscriptionId: string } };
+  disconnect: { type: 'disconnect'; payload: { subscriptionId: string } };
+  error: { type: 'error'; payload: Error };
+  message: { type: 'message'; payload: Bundle };
+  close: { type: 'close' };
+  heartbeat: { type: 'heartbeat'; payload: Bundle };
+};
+
+export type RobustWebSocketEventMap = {
```

Comment:
`Robust` 🤔 

Is this a nod to partysocket? 😛

Reply:
Possibly, its definitely inspired by `partysocket`